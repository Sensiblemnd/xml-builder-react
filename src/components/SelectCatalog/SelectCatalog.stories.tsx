import * as React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import SelectCatalog from "./SelectCatalog";

const options: Map<string, string> = new Map();
options.set("key1", "value1");
options.set("key2", "value2");
options.set("key3", "value3");

storiesOf("Select - Catalog", module)
  .add("Default", () => <SelectCatalog options={options} onChange={action} />)
  .add("Default value", () => (
    <SelectCatalog options={options} onChange={action} value={"key3"} />
  ))
  .add("Error", () => (
    <SelectCatalog options={options} onChange={action} error={true} />
  ));
