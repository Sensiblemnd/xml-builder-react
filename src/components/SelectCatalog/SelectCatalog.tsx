import React from "react";
import { FormSelect, FormSelectOption } from "@patternfly/react-core";

export interface ControlProps {
  value?: string;
  onChange?: (selected: string) => any;
}

export interface Props extends ControlProps {
  error?: any;
  options: Map<string, string>;
}

export interface State {
  selected: string;
}

class SelectCatalog extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      selected: props.value || ""
    };
  }

  handleOnChange = (value: string) => {
    const { onChange } = this.props;
    this.setState({ selected: value }, () => {
      if (onChange) {
        onChange(value);
      }
    });
  };

  render() {
    const { error, options } = this.props;
    const { selected } = this.state;

    return (
      <React.Fragment>
        <FormSelect
          value={selected}
          onChange={this.handleOnChange}
          isValid={!error}
          aria-label="Tipo comprobante Select"
        >
          {Array.from(options.keys()).map((key, index) => {
            const label = options.get(key) || "Invalid key";
            return <FormSelectOption key={index} value={key} label={label} />;
          })}
        </FormSelect>
      </React.Fragment>
    );
  }
}

export default SelectCatalog;
