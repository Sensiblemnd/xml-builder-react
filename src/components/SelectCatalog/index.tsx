import SelectCatalog from "./SelectCatalog";

export default SelectCatalog;
