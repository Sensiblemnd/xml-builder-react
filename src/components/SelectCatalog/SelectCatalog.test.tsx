import React from "react";
import { shallow } from "enzyme";
import SelectCatalog from "./SelectCatalog";

it("Renders without crashing", () => {
  const options: Map<string, string> = new Map();

  options.set("key1", "value1");
  options.set("key2", "value2");
  options.set("key3", "value3");

  const wrapper = shallow(<SelectCatalog options={options} />);

  expect(wrapper).toMatchSnapshot();
});
