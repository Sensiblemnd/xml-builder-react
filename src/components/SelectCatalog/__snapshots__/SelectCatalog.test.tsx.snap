// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<Fragment>
  <FormSelect
    aria-label="Tipo comprobante Select"
    className=""
    isDisabled={false}
    isRequired={false}
    isValid={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    validated="default"
    value=""
  >
    <FormSelectOption
      key="0"
      label="value1"
      value="key1"
    />
    <FormSelectOption
      key="1"
      label="value2"
      value="key2"
    />
    <FormSelectOption
      key="2"
      label="value3"
      value="key3"
    />
  </FormSelect>
</Fragment>
`;
